### ISSUE TITLE
bug - caches only html

### ISSUE THREAD

**Issue by Dexus77** [jiangwenyuan/nuster#42]
Six months later, I decided to look at Nuster again and, like the first time, I just spent my time and became disillusioned.  The potentially interesting project never got a normal installer and did not get rid of children's bugs.
 Apparently faster implement caching in HAProxy.

 During testing, the following bug was found; the latest version v2.1.2.19 does not cache anything but html.  For example, an attempt to cache js, css led to the addition of garbage in the cache until it was clogged by 80%.

```
global
  nuster cache on data-size 200m
  debug
frontend fe
  bind *:80
  default_backend be
backend be
  mode http
  nuster cache on
  nuster rule all
  server s1 192.168.111.211:8111
```

```[CACHE] on, data_size=209715200
-------------------------------------------------------------------------------------------
First GET
------------------------------------------------------------------------------------------
00000000:fe.accept(0004)=0008 from [127.0.0.1:56846] ALPN=<none>
00000000:be.clireq[0008:ffffffff]: GET /static/css/vendor/bootstrap.min.css HTTP/1.1
00000000:be.clihdr[0008:ffffffff]: User-Agent: curl/7.29.0
00000000:be.clihdr[0008:ffffffff]: Host: 127.0.0.1
00000000:be.clihdr[0008:ffffffff]: Accept: */*
[CACHE] Checking rule: all
[CACHE] Calculate key: method.scheme.host.uri.
[CACHE] Got key: GETHTTP127.0.0.1/static/css/vendor/bootstrap.min.css
[CACHE] Checking key existence: NOT EXIST
[CACHE] [REQ] Checking if rule pass: PASS
00000000:be.srvrep[0008:0009]: HTTP/1.1 200 OK
00000000:be.srvhdr[0008:0009]: Content-Type: text/css
00000000:be.srvhdr[0008:0009]: Last-Modified: Mon, 25 Jul 2016 12:53:28 GMT
00000000:be.srvhdr[0008:0009]: Content-Length: 121200
00000000:be.srvhdr[0008:0009]: Accept-Ranges: bytes
00000000:be.srvhdr[0008:0009]: SERVER: hdesign/0.0.1
00000000:be.srvhdr[0008:0009]: Cache-Control: max-age=0
00000000:be.srvhdr[0008:0009]: Expired: Wed, 01 May 2019 15:14:52 GMT
00000000:be.srvhdr[0008:0009]: Date: Thu, 02 May 2019 15:14:52 GMT
[CACHE] [RES] Checking status code: PASS
**[CACHE] To create**
00000001:be.clicls[adfd:ffffffff]
00000001:be.closed[adfd:ffffffff]
------------------------------------------------------------------------------------------
Sec GET
-----------------------------------------------------------------------------------------
00000002:fe.accept(0004)=0008 from [127.0.0.1:56850] ALPN=<none>
00000002:be.clireq[0008:ffffffff]: GET /static/css/vendor/bootstrap.min.css HTTP/1.1
00000002:be.clihdr[0008:ffffffff]: User-Agent: curl/7.29.0
00000002:be.clihdr[0008:ffffffff]: Host: 127.0.0.1
00000002:be.clihdr[0008:ffffffff]: Accept: */*
[CACHE] Checking rule: all
[CACHE] Calculate key: method.scheme.host.uri.
[CACHE] Got key: GETHTTP127.0.0.1/static/css/vendor/bootstrap.min.css
[CACHE] Checking key existence: NOT EXIST
[CACHE] [REQ] Checking if rule pass: PASS
00000002:be.srvrep[0008:0009]: HTTP/1.1 200 OK
00000002:be.srvhdr[0008:0009]: Content-Type: text/css
00000002:be.srvhdr[0008:0009]: Last-Modified: Mon, 25 Jul 2016 12:53:28 GMT
00000002:be.srvhdr[0008:0009]: Content-Length: 121200
00000002:be.srvhdr[0008:0009]: Accept-Ranges: bytes
00000002:be.srvhdr[0008:0009]: SERVER: hdesign/0.0.1
00000002:be.srvhdr[0008:0009]: Cache-Control: max-age=0
00000002:be.srvhdr[0008:0009]: Expired: Wed, 01 May 2019 15:15:51 GMT
00000002:be.srvhdr[0008:0009]: Date: Thu, 02 May 2019 15:15:51 GMT
[CACHE] [RES] Checking status code: PASS
**[CACHE] To create**
00000003:be.clicls[adfd:ffffffff]
00000003:be.closed[adfd:ffffffff]
```

As you can see, the shred creates correctly, but when it is repeated, it does not find it and creates again what and where it is incomprehensible, but the memory flows away.

 I say goodbye to this.

**Comment by packeteer** [comment]
with an attitude like that, you won't be missed!

**Comment by Dexus77** [comment]
If you are not able to create an elementary work product, do not do anything better.

**Comment by jiangwenyuan** [comment]
Hi @Dexus77 

Thanks for the report and sorry for the inconvenience. This is indeed a bug introduced in H19 branch.

BTW, v2.1.2.19 is development release and is unstable, you should use the stable release [v2.0.10.18](https://github.com/jiangwenyuan/nuster/releases/download/v2.0.10.18/nuster-2.0.10.18.tar.gz).

**Comment by Dexus77** [comment]
Hello!  Please forgive me, flared because time was spent on developing for Nuster and had to cancel and rewrite everything.

**Comment by jiangwenyuan** [comment]
Not at all, my bad. I should have tagged something more clear like `-dev`, sorry for wasting your time.
BTW, I've figured out this is due to some change in haproxy v1.9, and has been fixed in haproxy v2.0.
And I also work around this in current disk persistence branch, hope you could still be willing to help:)

**Comment by jiangwenyuan** [comment]
Fixed
